webapps/ui/cockpit/src/plugins/cockpit.processInstance.runtime.tab/userTasks/UserTasks.js

```diff
@@ -0,0 +1,242 @@
+    if (JSON.stringify(newQueryObject) !== JSON.stringify(queryObject)) {
+      setQueryObject(newQueryObject);
+    }
+  }, [
+    filter,
+    processInstanceId,
+    queryObject,
+    queryObject.activityInstanceIds,
+    sortOrder
+  ]);
+
+  // Fetch everything
+  useEffect(() => {
+    setLoadingState("LOADING");
+    post(
+      `%ENGINE_API%/task?maxResults=50&firstResult=${(page - 1) * 50}`,
+      queryObject
+    )
+      .then(res => res.json())
+      .then(setUserTasks)
+      .then(() => setLoadingState("DONE"));
+  }, [page, queryObject]);
+
+  // Fetch new Task count
+  useEffect(() => {
+    post("%ENGINE_API%/task/count", queryObject)
+      .then(res => res.json())
+      .then(json => setUserTasksCount(json.count));
+  }, [queryObject]);
+
+  if (loadingState === "LOADING") {
+    return <LoadingIndicator />;
+  }
+
+  if (!userTasks.length) {
+    return <span>No user tasks</span>;
+  }
+
+  const handleSortChange = property => {
+    const sortObj = {
+      sortBy: property,
+      sortOrder:
+        sortOrder.sortBy === property
+          ? sortOrder.sortOrder === "desc"
+            ? "asc"
+            : "desc"
+          : "asc"
+    };
+    setItem("sortPIUserTaskTab", sortObj);
+    setSortOrder(sortObj);
+  };
+
+  const handleSearch = activity => {
+    setFilter({
+      ...filter,
+      activityIds: [activity],
+      activityInstanceIds: [
+        activityIdToInstancesMap[activity.taskDefinitionKey].filter(execution =>
+          execution.executionIds.includes(activity.executionId)
+        )[0].id
+      ]
+    });
+  };
+
+  const rows = [
+    { searchQuery: "nameCaseInsensitive", label: "Activity" },
+    { searchQuery: "assignee", label: "Assignee" },
+    { label: "Owner" },
+    { searchQuery: "created", label: "Creation Date" },
+    { searchQuery: "dueDate", label: "Due Date" },
+    { searchQuery: "followUpDate", label: "Follow Up Date" },
+    { searchQuery: "priority", label: "Priority" },
+    { label: "Delegation State" },
+    { searchQuery: "id", label: "Task ID" },
+    { label: "Action" }
+  ];
+
+  return (
+    <>
+      <Table
+        className="UserTasks"
+        head={rows.map((heading, idx) => {
+          return (
+            <Table.Head
+              key={idx}
+              onSort={
+                heading.searchQuery
+                  ? () => handleSortChange(heading.searchQuery)
+                  : null
+              }
+              sortOrder={
+                sortOrder.sortBy === heading.searchQuery
+                  ? sortOrder.sortOrder
+                  : null
+              }
+            >
+              {heading.label}
+            </Table.Head>
+          );
+        })}
+      >
+        {userTasks.map((row, idx) => {
+          return (
+            <Table.Row key={idx}>
+              <Table.Cell>
+                <LinkButton onClick={() => handleSearch(row)}>
+                  {bpmnElements[row.taskDefinitionKey]?.name ||
```

Comment:
💚 
I wasn't aware that the `?.` operator made it to JavaScript. I already knew it from Groovy.

Reply:
Yeah, but our linter does not fully support it for functions yet. So `element.function?.()` does not work